lib/datastore/request.js

```diff
@@ -617,11 +616,15 @@ DatastoreRequest.prototype.runQuery = function(query, callback) {
       return;
     }
 
-    var entities = entity.formatArray(resp.batch.entity_result);
+    var entities = [];
     var nextQuery = null;
 
-    if (resp.batch.end_cursor && entities.length > 0) {
-      var endCursor = resp.batch.end_cursor.toBase64();
+    if (resp.batch.entityResults) {
+      entities = entity.formatArray(resp.batch.entityResults);
+    }
+
+    if (resp.batch.endCursor && entities.length > 0) {
+      var endCursor = resp.batch.endCursor.toBase64();
 
       nextQuery = extend(true, new Query(), query);
       nextQuery.start(endCursor).offset(0);
```

Comment:
You aren't guaranteed that we'll completely answer the offset. Instead you should compare skipped_results to the initial offset.